crl/updater/updater_test.go

```diff
@@ -1,12 +1,288 @@
 package updater
 
 import (
+	"context"
+	"errors"
+	"io"
 	"testing"
 	"time"
 
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/jmhodges/clock"
+	capb "github.com/letsencrypt/boulder/ca/proto"
+	corepb "github.com/letsencrypt/boulder/core/proto"
+	cspb "github.com/letsencrypt/boulder/crl/storer/proto"
+	"github.com/letsencrypt/boulder/issuance"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/metrics"
+	"github.com/letsencrypt/boulder/mocks"
+	sapb "github.com/letsencrypt/boulder/sa/proto"
 	"github.com/letsencrypt/boulder/test"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
+// fakeGRCC is a fake sapb.StorageAuthority_GetRevokedCertsClient which can be
+// populated with some CRL entries or an error for use as the return value of
+// a faked GetRevokedCerts call.
+type fakeGRCC struct {
+	grpc.ClientStream
+	entries []*corepb.CRLEntry
+	nextIdx int
+	err     error
+}
+
+func (f *fakeGRCC) Recv() (*corepb.CRLEntry, error) {
+	if f.err != nil {
+		return nil, f.err
+	}
+	if f.nextIdx < len(f.entries) {
+		res := f.entries[f.nextIdx]
+		f.nextIdx++
+		return res, nil
+	}
+	return nil, io.EOF
+}
+
+// fakeSAC is a fake sapb.StorageAuthorityClient which can be populated with a
+// fakeGRCC to be used as the return value for calls to GetRevokedCerts.
+type fakeSAC struct {
+	mocks.StorageAuthority
+	grcc fakeGRCC
+}
+
+func (f *fakeSAC) GetRevokedCerts(ctx context.Context, _ *sapb.GetRevokedCertsRequest, _ ...grpc.CallOption) (sapb.StorageAuthority_GetRevokedCertsClient, error) {
+	return &f.grcc, nil
+}
+
+// fakeGCC is a fake capb.CRLGenerator_GenerateCRLClient which can be
+// populated with some CRL entries or an error for use as the return value of
+// a faked GenerateCRL call.
+type fakeGCC struct {
+	grpc.ClientStream
+	chunks  [][]byte
+	nextIdx int
+	sendErr error
+	recvErr error
+}
+
+func (f *fakeGCC) Send(*capb.GenerateCRLRequest) error {
+	if f.sendErr != nil {
+		return f.sendErr
+	}
+	return nil
```

Comment:
Seems like you could simplify this to `return f.sendErr` :-)

(and below)

Reply:
Explained in https://github.com/letsencrypt/boulder/pull/6382#discussion_r971228125 - it's safer to have the named return so we know it's always assigned to.

Reply:
Good catch! Done.